In [1]:
import pandas as pd
import numpy as np

In [2]:
m16 = pd.read_csv("clean2016Olympics_Men's_artistic_team_all-around.csv")

In [3]:
m16.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 15 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Team        8 non-null      object 
 1   fx          8 non-null      float64
 2   fx_rank     8 non-null      float64
 3   ph          8 non-null      float64
 4   ph_rank     8 non-null      float64
 5   sr          8 non-null      float64
 6   sr_rank     8 non-null      float64
 7   vt          8 non-null      float64
 8   vt_rank     8 non-null      float64
 9   pb          8 non-null      float64
 10  pb_rank     8 non-null      float64
 11  hb          8 non-null      float64
 12  hb_rank     8 non-null      float64
 13  total       8 non-null      float64
 14  total_rank  8 non-null      float64
dtypes: float64(14), object(1)
memory usage: 1.1+ KB


In [4]:
m16

Team      fx  fx_rank      ph  ph_rank      sr  sr_rank      vt  \
0          Japan  47.199      1.0  43.933      4.0  44.599      3.0  46.199   
1         Russia  44.766      3.0  45.299      2.0  45.432      1.0  46.033   
2          China  43.799      4.0  44.258      3.0  45.233      2.0  45.000   
3  Great Britain  45.099      2.0  45.623      1.0  44.066      6.0  45.399   
4  United States  43.757      5.0  43.699      5.0  44.465      4.0  45.865   
5         Brazil  41.733      7.0  43.433      7.0  44.332      5.0  45.032   
6        Germany  43.352      6.0  40.948      8.0  43.132      7.0  44.540   
7        Ukraine  27.766      8.0  43.632      6.0  29.299      8.0  44.015   

   vt_rank      pb  pb_rank      hb  hb_rank    total  total_rank  
0      1.0  46.766      2.0  45.398      1.0  274.094         1.0  
1      2.0  46.033      4.0  43.890      6.0  271.453         2.0  
2      6.0  47.866      1.0  44.966      3.0  271.122         3.0  
3      4.0  44.566      6.0  44.999      2.0  269.752         4.0  
4      3.0  46.333      3.0  44.441      5.0  268.560         5.0  
5      5.0  44.533      6.0  44.665      4.0  263.728         6.0  
6      7.0  45.391      7.0  43.732      7.0  261.275         7.0  
7      8.0  30.366      8.0  27.000      8.0  202.078         8.0

In [5]:
m20 = pd.read_csv("clean2020Olympics_Men's_artistic_team_all-around.csv")

In [6]:
m20

Rank           Team      fx  fx_rank      ph  ph_rank      sr  sr_rank  \
0   1.0            ROC  42.632        2  43.140        4  44.399        1   
1   2.0          Japan  43.700        1  43.566        3  42.433        3   
2   3.0          China  42.132        4  43.966        1  43.599        2   
3   4.0  Great Britain  42.432        3  43.632        2  41.833        5   
4   5.0  United States  39.965        7  42.099        5  42.166        4   
5   6.0    Switzerland  42.066        5  41.099        6  40.298        8   
6   7.0        Ukraine  41.232        6  39.999        7  40.866        6   
7   8.0        Germany  34.832        8  39.200        8  40.333        7   

       vt  vt_rank      pb  pb_rank      hb  hb_rank    Total  
0  44.765        1  45.099        2  42.465        4  262.500  
1  44.232        3  44.666        3  43.800        1  262.397  
2  44.332        2  45.199        1  42.666        3  261.894  
3  43.032        4  42.932        6  41.899        5  255.760  
4  42.899        5  44.266        4  43.199        2  254.594  
5  41.999        7  44.066        5  41.399        6  250.927  
6  41.499        8  42.233        8  40.565        7  246.394  
7  42.366        6  42.765        7  38.999        8  238.495

In [7]:
m12 = pd.read_csv("clean2012Olympics_Men's_artistic_team_all-around.csv")

In [78]:
m12

Rank           Team      fx  fx_rank      ph  ph_rank      sr  sr_Rank  \
0     1          China  45.133        5  43.833        4  44.899        5   
1     2          Japan  44.733        6  42.365        5  45.699        2   
2     3  Great Britain  46.132        1  45.932        1  43.066        8   
3     4        Ukraine  44.065        8  44.966        3  45.699        2   
4     5  United States  45.266        3  40.633        7  45.257        4   
5     6         Russia  45.308        2  42.299        6  46.132        1   
6     7        Germany  45.199        4  40.466        8  44.824        6   
7     8         France  44.066        7  45.265        2  43.432        7   

       vt  vt_rank      pb  pb_Rank      hb  hb_rank    Total  year  
0  48.316        1  46.750        1  47.066        1  275.997  2012  
1  45.971        7  46.282        2  46.899        2  271.952  2012  
2  48.182        2  44.566        7  43.833        7  271.711  2012  
3  48.065        3  44.733        6  43.998        6  271.526  2012  
4  46.632        6  45.765        3  46.399        3  269.952  2012  
5  46.866        5  43.965        8  45.033        5  269.603  2012  
6  47.032        4  45.433        4  45.065        4  268.019  2012  
7  44.473        8  45.290        5  42.915        8  265.441  2012

In [9]:
m16.columns

Index(['Team', 'fx', 'fx_rank', 'ph', 'ph_rank', 'sr', 'sr_rank', 'vt',
       'vt_rank', 'pb', 'pb_rank', 'hb', 'hb_rank', 'total', 'total_rank'],
      dtype='object')

In [10]:
column_order = ['total_rank', 'Team', 'fx', 'fx_rank', 'ph', 'ph_rank', 'sr', 'sr_rank', 'vt',
       'vt_rank', 'pb', 'pb_rank', 'hb', 'hb_rank', 'total', ]

# Create a new DataFrame with the desired column order
m16 = m16[column_order]

In [11]:
m16

total_rank           Team      fx  fx_rank      ph  ph_rank      sr  \
0         1.0          Japan  47.199      1.0  43.933      4.0  44.599   
1         2.0         Russia  44.766      3.0  45.299      2.0  45.432   
2         3.0          China  43.799      4.0  44.258      3.0  45.233   
3         4.0  Great Britain  45.099      2.0  45.623      1.0  44.066   
4         5.0  United States  43.757      5.0  43.699      5.0  44.465   
5         6.0         Brazil  41.733      7.0  43.433      7.0  44.332   
6         7.0        Germany  43.352      6.0  40.948      8.0  43.132   
7         8.0        Ukraine  27.766      8.0  43.632      6.0  29.299   

   sr_rank      vt  vt_rank      pb  pb_rank      hb  hb_rank    total  
0      3.0  46.199      1.0  46.766      2.0  45.398      1.0  274.094  
1      1.0  46.033      2.0  46.033      4.0  43.890      6.0  271.453  
2      2.0  45.000      6.0  47.866      1.0  44.966      3.0  271.122  
3      6.0  45.399      4.0  44.566      6.0  44.999      2.0  269.752  
4      4.0  45.865      3.0  46.333      3.0  44.441      5.0  268.560  
5      5.0  45.032      5.0  44.533      6.0  44.665      4.0  263.728  
6      7.0  44.540      7.0  45.391      7.0  43.732      7.0  261.275  
7      8.0  44.015      8.0  30.366      8.0  27.000      8.0  202.078

In [12]:
column_order = [ 'total_Rank', 'Team', 'fx', 'fx_Rank', 'ph', 'ph_Rank', 'sr', 'sr_Rank', 'vt',
       'vt_Rank', 'pb', 'pb_Rank', 'hb', 'hb_Rank', 'tota;',]

m12= m12[column_order]

In [13]:
m12.columns

Index(['total_Rank', 'Team', 'fx', 'fx_Rank', 'ph', 'ph_Rank', 'sr', 'sr_Rank',
       'vt', 'vt_Rank', 'pb', 'pb_Rank', 'hb', 'hb_Rank', 'tota;'],
      dtype='object')

In [14]:
m16.rename(columns={'total_rank': 'Rank'}, inplace=True)

In [15]:
m16.rename(columns={'total': 'Total'}, inplace=True)

In [16]:
m12.head()

total_Rank           Team      fx  fx_Rank      ph  ph_Rank      sr  \
0         1.0          China  45.133      5.0  43.833      4.0  44.899   
1         2.0          Japan  44.733      6.0  42.365      5.0  45.699   
2         3.0  Great Britain  46.132      1.0  45.932      1.0  43.066   
3         4.0        Ukraine  44.065      8.0  44.966      3.0  45.699   
4         5.0  United States  45.266      3.0  40.633      7.0  45.257   

   sr_Rank      vt  vt_Rank      pb  pb_Rank      hb  hb_Rank    tota;  
0      5.0  48.316      1.0  46.750      1.0  47.066      1.0  275.997  
1      2.0  45.971      7.0  46.282      2.0  46.899      2.0  271.952  
2      8.0  48.182      2.0  44.566      7.0  43.833      7.0  271.711  
3      2.0  48.065      3.0  44.733      6.0  43.998      6.0  271.526  
4      4.0  46.632      6.0  45.765      3.0  46.399      3.0  269.952

In [26]:
m12.rename(columns={'total_Rank': 'Rank'}, inplace=True)

In [18]:
m12.rename(columns={'tota;': 'Total'}, inplace=True)

In [72]:
m12.rename(columns={'fx_Rank': 'fx_rank'}, inplace=True)

In [73]:
m12.rename(columns={'ph_Rank': 'ph_rank'}, inplace=True)

In [74]:
m12.rename(columns={'vt_Rank': 'vt_rank'}, inplace=True)

In [75]:
m12.rename(columns={'hb_Rank': 'hb_rank'}, inplace=True)

In [79]:
m12.rename(columns={'sr_Rank': 'sr_rank'}, inplace=True)

In [80]:
m12.rename(columns={'pb_Rank': 'pb_rank'}, inplace=True)

In [56]:
m16['year'] = 2016

In [57]:
m20['year'] = 2020

In [58]:
m12['year'] = 2012

In [59]:
m20.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 16 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Rank     8 non-null      float64
 1   Team     8 non-null      object 
 2   fx       8 non-null      float64
 3   fx_rank  8 non-null      int64  
 4   ph       8 non-null      float64
 5   ph_rank  8 non-null      int64  
 6   sr       8 non-null      float64
 7   sr_rank  8 non-null      int64  
 8   vt       8 non-null      float64
 9   vt_rank  8 non-null      int64  
 10  pb       8 non-null      float64
 11  pb_rank  8 non-null      int64  
 12  hb       8 non-null      float64
 13  hb_rank  8 non-null      int64  
 14  Total    8 non-null      float64
 15  year     8 non-null      int64  
dtypes: float64(8), int64(7), object(1)
memory usage: 1.1+ KB


In [60]:
m16.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 16 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Rank     8 non-null      int64  
 1   Team     8 non-null      object 
 2   fx       8 non-null      float64
 3   fx_rank  8 non-null      int64  
 4   ph       8 non-null      float64
 5   ph_rank  8 non-null      int64  
 6   sr       8 non-null      float64
 7   sr_rank  8 non-null      int64  
 8   vt       8 non-null      float64
 9   vt_rank  8 non-null      int64  
 10  pb       8 non-null      float64
 11  pb_rank  8 non-null      int64  
 12  hb       8 non-null      float64
 13  hb_rank  8 non-null      int64  
 14  Total    8 non-null      float64
 15  year     8 non-null      int64  
dtypes: float64(7), int64(8), object(1)
memory usage: 1.1+ KB


In [61]:
newdf = m16.copy()

In [62]:
newdf

Rank           Team      fx  fx_rank      ph  ph_rank      sr  sr_rank  \
0     1          Japan  47.199        1  43.933        4  44.599        3   
1     2         Russia  44.766        3  45.299        2  45.432        1   
2     3          China  43.799        4  44.258        3  45.233        2   
3     4  Great Britain  45.099        2  45.623        1  44.066        6   
4     5  United States  43.757        5  43.699        5  44.465        4   
5     6         Brazil  41.733        7  43.433        7  44.332        5   
6     7        Germany  43.352        6  40.948        8  43.132        7   
7     8        Ukraine  27.766        8  43.632        6  29.299        8   

       vt  vt_rank      pb  pb_rank      hb  hb_rank    Total  year  
0  46.199        1  46.766        2  45.398        1  274.094  2016  
1  46.033        2  46.033        4  43.890        6  271.453  2016  
2  45.000        6  47.866        1  44.966        3  271.122  2016  
3  45.399        4  44.566        6  44.999        2  269.752  2016  
4  45.865        3  46.333        3  44.441        5  268.560  2016  
5  45.032        5  44.533        6  44.665        4  263.728  2016  
6  44.540        7  45.391        7  43.732        7  261.275  2016  
7  44.015        8  30.366        8  27.000        8  202.078  2016

In [63]:
# Apply the function to selected columns using apply and lambda
newdf = newdf.apply(lambda col: col.astype(int) if 'rank' in col.name.lower() else col)


In [64]:
newdf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 16 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Rank     8 non-null      int64  
 1   Team     8 non-null      object 
 2   fx       8 non-null      float64
 3   fx_rank  8 non-null      int64  
 4   ph       8 non-null      float64
 5   ph_rank  8 non-null      int64  
 6   sr       8 non-null      float64
 7   sr_rank  8 non-null      int64  
 8   vt       8 non-null      float64
 9   vt_rank  8 non-null      int64  
 10  pb       8 non-null      float64
 11  pb_rank  8 non-null      int64  
 12  hb       8 non-null      float64
 13  hb_rank  8 non-null      int64  
 14  Total    8 non-null      float64
 15  year     8 non-null      int64  
dtypes: float64(7), int64(8), object(1)
memory usage: 1.1+ KB


In [65]:
m16=newdf.copy()

In [66]:
# Apply the function to selected columns using apply and lambda
m12 = m12.apply(lambda col: col.astype(int) if 'rank' in col.name.lower() else col)


In [67]:
m12.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 16 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Rank     8 non-null      int64  
 1   Team     8 non-null      object 
 2   fx       8 non-null      float64
 3   fx_Rank  8 non-null      int64  
 4   ph       8 non-null      float64
 5   ph_Rank  8 non-null      int64  
 6   sr       8 non-null      float64
 7   sr_Rank  8 non-null      int64  
 8   vt       8 non-null      float64
 9   vt_Rank  8 non-null      int64  
 10  pb       8 non-null      float64
 11  pb_Rank  8 non-null      int64  
 12  hb       8 non-null      float64
 13  hb_Rank  8 non-null      int64  
 14  Total    8 non-null      float64
 15  year     8 non-null      int64  
dtypes: float64(7), int64(8), object(1)
memory usage: 1.1+ KB


In [81]:
combined_df = pd.concat([m20, m16, m12], ignore_index=True)

In [82]:
combined_df

Rank           Team      fx  fx_rank      ph  ph_rank      sr  sr_rank  \
0    1.0            ROC  42.632        2  43.140        4  44.399        1   
1    2.0          Japan  43.700        1  43.566        3  42.433        3   
2    3.0          China  42.132        4  43.966        1  43.599        2   
3    4.0  Great Britain  42.432        3  43.632        2  41.833        5   
4    5.0  United States  39.965        7  42.099        5  42.166        4   
5    6.0    Switzerland  42.066        5  41.099        6  40.298        8   
6    7.0        Ukraine  41.232        6  39.999        7  40.866        6   
7    8.0        Germany  34.832        8  39.200        8  40.333        7   
8    1.0          Japan  47.199        1  43.933        4  44.599        3   
9    2.0         Russia  44.766        3  45.299        2  45.432        1   
10   3.0          China  43.799        4  44.258        3  45.233        2   
11   4.0  Great Britain  45.099        2  45.623        1  44.066        6   
12   5.0  United States  43.757        5  43.699        5  44.465        4   
13   6.0         Brazil  41.733        7  43.433        7  44.332        5   
14   7.0        Germany  43.352        6  40.948        8  43.132        7   
15   8.0        Ukraine  27.766        8  43.632        6  29.299        8   
16   1.0          China  45.133        5  43.833        4  44.899        5   
17   2.0          Japan  44.733        6  42.365        5  45.699        2   
18   3.0  Great Britain  46.132        1  45.932        1  43.066        8   
19   4.0        Ukraine  44.065        8  44.966        3  45.699        2   
20   5.0  United States  45.266        3  40.633        7  45.257        4   
21   6.0         Russia  45.308        2  42.299        6  46.132        1   
22   7.0        Germany  45.199        4  40.466        8  44.824        6   
23   8.0         France  44.066        7  45.265        2  43.432        7   

        vt  vt_rank      pb  pb_rank      hb  hb_rank    Total  year  
0   44.765        1  45.099        2  42.465        4  262.500  2020  
1   44.232        3  44.666        3  43.800        1  262.397  2020  
2   44.332        2  45.199        1  42.666        3  261.894  2020  
3   43.032        4  42.932        6  41.899        5  255.760  2020  
4   42.899        5  44.266        4  43.199        2  254.594  2020  
5   41.999        7  44.066        5  41.399        6  250.927  2020  
6   41.499        8  42.233        8  40.565        7  246.394  2020  
7   42.366        6  42.765        7  38.999        8  238.495  2020  
8   46.199        1  46.766        2  45.398        1  274.094  2016  
9   46.033        2  46.033        4  43.890        6  271.453  2016  
10  45.000        6  47.866        1  44.966        3  271.122  2016  
11  45.399        4  44.566        6  44.999        2  269.752  2016  
12  45.865        3  46.333        3  44.441        5  268.560  2016  
13  45.032        5  44.533        6  44.665        4  263.728  2016  
14  44.540        7  45.391        7  43.732        7  261.275  2016  
15  44.015        8  30.366        8  27.000        8  202.078  2016  
16  48.316        1  46.750        1  47.066        1  275.997  2012  
17  45.971        7  46.282        2  46.899        2  271.952  2012  
18  48.182        2  44.566        7  43.833        7  271.711  2012  
19  48.065        3  44.733        6  43.998        6  271.526  2012  
20  46.632        6  45.765        3  46.399        3  269.952  2012  
21  46.866        5  43.965        8  45.033        5  269.603  2012  
22  47.032        4  45.433        4  45.065        4  268.019  2012  
23  44.473        8  45.290        5  42.915        8  265.441  2012

In [100]:
# combined_df.to_csv('cleaned2012to2020mens.csv', index=False)

In [83]:
filtered_df = combined_df[combined_df['Rank'].isin([1, 2, 3])]
filtered_df

Rank           Team      fx  fx_rank      ph  ph_rank      sr  sr_rank  \
0    1.0            ROC  42.632        2  43.140        4  44.399        1   
1    2.0          Japan  43.700        1  43.566        3  42.433        3   
2    3.0          China  42.132        4  43.966        1  43.599        2   
8    1.0          Japan  47.199        1  43.933        4  44.599        3   
9    2.0         Russia  44.766        3  45.299        2  45.432        1   
10   3.0          China  43.799        4  44.258        3  45.233        2   
16   1.0          China  45.133        5  43.833        4  44.899        5   
17   2.0          Japan  44.733        6  42.365        5  45.699        2   
18   3.0  Great Britain  46.132        1  45.932        1  43.066        8   

        vt  vt_rank      pb  pb_rank      hb  hb_rank    Total  year  
0   44.765        1  45.099        2  42.465        4  262.500  2020  
1   44.232        3  44.666        3  43.800        1  262.397  2020  
2   44.332        2  45.199        1  42.666        3  261.894  2020  
8   46.199        1  46.766        2  45.398        1  274.094  2016  
9   46.033        2  46.033        4  43.890        6  271.453  2016  
10  45.000        6  47.866        1  44.966        3  271.122  2016  
16  48.316        1  46.750        1  47.066        1  275.997  2012  
17  45.971        7  46.282        2  46.899        2  271.952  2012  
18  48.182        2  44.566        7  43.833        7  271.711  2012

In [84]:
df = filtered_df.copy()

In [85]:
df

Rank           Team      fx  fx_rank      ph  ph_rank      sr  sr_rank  \
0    1.0            ROC  42.632        2  43.140        4  44.399        1   
1    2.0          Japan  43.700        1  43.566        3  42.433        3   
2    3.0          China  42.132        4  43.966        1  43.599        2   
8    1.0          Japan  47.199        1  43.933        4  44.599        3   
9    2.0         Russia  44.766        3  45.299        2  45.432        1   
10   3.0          China  43.799        4  44.258        3  45.233        2   
16   1.0          China  45.133        5  43.833        4  44.899        5   
17   2.0          Japan  44.733        6  42.365        5  45.699        2   
18   3.0  Great Britain  46.132        1  45.932        1  43.066        8   

        vt  vt_rank      pb  pb_rank      hb  hb_rank    Total  year  
0   44.765        1  45.099        2  42.465        4  262.500  2020  
1   44.232        3  44.666        3  43.800        1  262.397  2020  
2   44.332        2  45.199        1  42.666        3  261.894  2020  
8   46.199        1  46.766        2  45.398        1  274.094  2016  
9   46.033        2  46.033        4  43.890        6  271.453  2016  
10  45.000        6  47.866        1  44.966        3  271.122  2016  
16  48.316        1  46.750        1  47.066        1  275.997  2012  
17  45.971        7  46.282        2  46.899        2  271.952  2012  
18  48.182        2  44.566        7  43.833        7  271.711  2012

In [101]:
# df.to_csv('cleaned2012to2020mens_medaled.csv', index=False)

In [102]:
max_total = df['Total'].max()

In [96]:
max_total

275.997

In [97]:
min_total = df['Total'].min()

In [98]:
min_total

261.894

# In theory a team will need to score 261.894 or higher to place in top 3 
